merkle/map_verifier.go

```diff
@@ -42,23 +42,44 @@ func VerifyMapInclusionProof(treeID int64, index, leafHash, expectedRoot []byte,
 		}
 	}
 
+	var runningHash []byte
+	if len(leaf) != 0 {
+		runningHash = h.HashLeaf(treeID, index, leaf)
+	}
+
 	nID := storage.NewNodeIDFromHash(index)
+	for height, sib := range nID.Siblings() {
+		pElement := proof[height]
 
-	runningHash := make([]byte, len(leafHash))
-	copy(runningHash, leafHash)
+		// Since HashChildren(e0, e0) is not always equal to the empty
```

Comment:
Basically OK. Looks like there might be a couple of typos "leve1" and the HashChildren line
seems to be truncated.